src/Compilers/CSharp/Portable/BoundTree/BoundNode.cs

```diff
@@ -102,11 +102,20 @@ public bool HasAnyErrors
         {
             get
             {
-                // NOTE: check Syntax rather than WasCompilerGenerated because sequence points can have null syntax.
-                if (this.HasErrors || this.Syntax != null && this.Syntax.HasErrors)
+                if (this.HasErrors)
                 {
                     return true;
                 }
+
+                // The syntax attached to placeholders should not limit usage of those placeholders
+                if (this is not BoundValuePlaceholderBase)
```

Comment:
>if (this is not BoundValuePlaceholderBase [](http://example.com/codeflow?start=16&length=41)

While I do not object to making this change, it feels like this is not a fix for the problem the PR is claiming to address. In order for SemanticModel to work properly, every syntax node that is normally bindable should be bound, regardless of errors, That means that the value returned by this helper shouldn't matter. The lambda should be bound even if this helper returns true.

Reply:
We might suppress additional diagnostics based on what this helper returns, but we should not be suppressing the binding.

Reply:
I can adjust the PR title if that's what you're suggesting. Like in another lambda-related PR last week, the question is whether we do `BindToTypeForErrorRecovery` (which results in poor lambda binding) or `GenerateConversionForAssignment` (which binds a conversion so results in better lambda information).
Maybe "Adjust lambda binding in object initializer despite errors"?